middle_end/flambda/basic/simple.ml

```diff
@@ -55,23 +55,28 @@ let const_from_descr descr = const (RWC.of_descr descr)
 
 let without_coercion t = pattern_match t ~name ~const
 
-let merge_coercion t ~newer_coercion =
-  if is_const t then None
+let apply_coercion t applied_coercion =
+  if Coercion.is_id applied_coercion then Some t
   else
-    match newer_coercion with
-    | None -> Some t
-    | Some newer_coercion ->
-      let coercion =
-        match coercion t with
-        | None -> newer_coercion
-        | Some older_coercion ->
-          Rec_info.merge older_coercion ~newer:newer_coercion
-      in
-      Some (with_coercion (without_coercion t) coercion)
+    let coercion = 
```

Comment:
Why not just use `Coercion.compose` here? The check to `Coercion.is_id` will be done by it anyway.

Reply:
Ah, good point.